apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,206 @@
+      };
+    }
+
+    const {formUrl, authTokenName, authTokenValue} = this.props.scriptData;
+    $.post({
+      url: formUrl + '.json',
+      dataType: "json",
+      data: {
+        '_method': 'patch',
+        [authTokenName]: authTokenValue,
+        "user[school_info_attributes][country]": this.state.country,
+        "user[school_info_attributes][school_type]": this.state.schoolType,
+        ...schoolData,
+      },
+    }).done(() => {
+      this.props.onClose();
+    }).fail(() => {
+      if (!this.state.showSchoolInfoUnknownError) {
+        // First failure, display error message and give the teacher a chance
+        // to try again.
+        this.setState({showSchoolInfoUnknownError: true});
+      } else {
+        // We already failed once, let's not block the teacher any longer.
+        // TODO (bbuchanan): Gather metrics on this.
+        this.props.onClose();
+      }
+    });
+  };
+
+  onCountryChange = (_, event) => {
+    const newCountry = event ? event.value : '';
+    this.setState({country: newCountry});
+  };
+
+  onSchoolTypeChange = (event) => {
+    const newType = event ? event.target.value : '';
+    this.setState({schoolType: newType});
+  };
+
+  onSchoolChange = (_, event) => {
+    const newSchool = event ? event.value : '';
+    this.setState({ncesSchoolId: newSchool});
+  };
+
+  onSchoolNotFoundChange = (field, event) => {
+    let newValue = event ? event.target.value : '';
+    this.setState({
+      [field]: newValue
+    });
+  };
+
+  render() {
+    return (
+      <BaseDialog
+        useUpdatedStyles
+        isOpen={true}
+        handleClose={this.props.onClose}
+        uncloseable
+      >
+        <div style={styles.container}>
+          <div style={styles.heading}>
+            We want to bring Computer Science to every student - help us track our progress!
+          </div>
+          {this.state.showSchoolInfoUnknownError && (
+            <p style={styles.error}>We encountered an error with your submission. Please try again.</p>
+          )}
+          <div style={styles.middle}>
+            <p>
+              Please enter your school information below.
+            </p>
+            <SchoolInfoInputs
+              ref={ref => this.schoolInfoInputs = ref}
+              onCountryChange={this.onCountryChange}
+              onSchoolTypeChange={this.onSchoolTypeChange}
+              onSchoolChange={this.onSchoolChange}
+              onSchoolNotFoundChange={this.onSchoolNotFoundChange}
+              country={this.state.country}
+              schoolType={this.state.schoolType}
+              ncesSchoolId={this.state.ncesSchoolId}
+              schoolName={this.state.schoolName}
+              schoolState={this.state.schoolState}
+              schoolZip={this.state.schoolZip}
+              schoolLocation={this.state.schoolLocation}
+              useGoogleLocationSearch={false}
+              showErrors={false}
```

Comment:
Does the `schoolLocation` value even get used if `useGoogleLocationSearch` is false?